homeassistant/components/filter/sensor.py

```diff
@@ -466,7 +464,8 @@ def filter_state(self, new_state: _State) -> _State:
             raise ValueError(f"State <{fstate.state}> is not a Number")
 
         filtered = self._filter_state(fstate)
-        filtered.set_precision(self.filter_precision)
+        if self.filter_precision is not None:
+            filtered.set_precision(self.filter_precision)
```

Comment:
Is this change necessary?
Does it not end up being a duplicate of the change on line 384/386?

Reply:
I believe so, filter_precision starts as None

Reply:
What I mean is that the first line in set_precision is check that the passed value is not None.
`if precision is not None and isinstance(self.state, Number):`

So you are first checking that it's not None, and then checking a second time for the same thing.

Reply:
Sorry misunderstood initial comment (fixed)